basics_of_sst-core/tutorial/Example04.md

```diff
@@ -0,0 +1,488 @@
+
+1.  Navigate to the `example` folder and create a folder named `Example04`.
+2.  Copy the `Example03` files to the `Example04` folder.
+3.  Open the copied `ExampleComponent.h` file and change the namespace to `Example04`.
+4.  Open the copied `ExampleComponent.cc` file and change the `using namespace` directive to refer to the `Example04` namespace.
+5.  Open the copied `ExampleEvent.h` file and change the namespace to `Example04`.
+
+## Creating a SubComponent
+
+Creating a subcomponent is similar to creating a regular component.  First, navigate to the Example04 folder and, using the editor of your choice, create an new file and enter the code shown below.  Save it as `ExampleSubComponent.h`.
+
+```
+#ifndef EXAMPLE_SUBCOMPONENT_H
+#define EXAMPLE_SUBCOMPONENT_H
+
+#include <sst/core/subcomponent.h>
+#include <sst/core/output.h>
+
+const uint64_t FATAL = 0;
+const uint64_t WARN  = 1;
+const uint64_t INFO  = 2;
+const uint64_t DEBUG = 3;
+const uint64_t TRACE = 4;
+const uint64_t ALL   = 5;
+
+using namespace SST;
+
+namespace Example04 
+{
+    class ExampleSubComponent : public SST::SubComponent
+    {
+        public:
+            // Legacy constructors included for background compatibility.
+            // Needed for V9, can be removed for V10.
+            //
+            ExampleSubComponent(Component *owningComponent) :
+                SubComponent(owningComponent)
+            { }
+            ExampleSubComponent(Component* comp, Params& params) :
+                SubComponent(comp)
+            { }
+
+            // Updated constructors.  No longer pass in a parent.
+            // Instead, use a component id.
+            //
+            ExampleSubComponent(ComponentId_t id) :
+                SubComponent(id)
+            { }
+            ExampleSubComponent(ComponentId_t id, Params& params) :
+                SubComponent(id)
+            { }
+
+            virtual ~ExampleSubComponent() {}
+            
+            // Clock handler.  This is the method called from the clock event.
+            //
+            virtual void clock(Cycle_t) {}
+
+            // Subcomponent public method for returning current count.
+            //
+            int getCount() { return count_++; }
+
+            // Register the subcomponent API.
+            //
+            SST_ELI_REGISTER_SUBCOMPONENT_API(Example04::ExampleSubComponent)
+
+            // Register the subcomponent.
+            //
+            SST_ELI_REGISTER_SUBCOMPONENT(
+                ExampleSubComponent,                // INSERT_CLASS_NAME
+                "example",                          // Element library.  Name of the .so.
+                "ExampleSubComponent",              // "INSERT_CLASS_NAME"
+                SST_ELI_ELEMENT_VERSION(1, 0, 0),   // 
+                "Example subcomponent",             // Subcomponent description
+                "SST::SubComponent"                 // "INSERT_FULL_PARENT_CLASS_NAME" or 
```

Comment:
This should be Example04::ExampleSubComponent, as it is not the parent class but the API that this SubComponent implements (which in this case is itself).